src/commonMain/kotlin/fr/acinq/lightning/channel/Commitments.kt

```diff
@@ -421,7 +422,18 @@ data class Commitment(
             "built remote commit number=${remoteCommit.index + 1} toLocalMsat=${spec.toLocal.toLong()} toRemoteMsat=${spec.toRemote.toLong()} htlc_in=$htlcsIn htlc_out=$htlcsOut feeratePerKw=${spec.feerate} txId=${remoteCommitTx.tx.txid} fundingTxId=$fundingTxId"
         }
 
-        val commitSig = CommitSig(params.channelId, sig, htlcSigs.toList())
+        val commitSig = when (spec.htlcs.size) {
+            0 -> {
+                val alternativeSigs = Commitments.alternativeFeerates.map { feerate ->
+                    val alternativeSpec = spec.copy(feerate = feerate)
+                    val (alternativeRemoteCommitTx, _) = Commitments.makeRemoteTxs(channelKeys, commitTxNumber = remoteCommit.index + 1, params.localParams, params.remoteParams, fundingTxIndex = fundingTxIndex, remoteFundingPubKey = remoteFundingPubkey, commitInput, remotePerCommitmentPoint = remoteNextPerCommitmentPoint, alternativeSpec)
+                    val alternativeSig = Transactions.sign(alternativeRemoteCommitTx, channelKeys.fundingKey(fundingTxIndex))
+                    CommitSigTlv.AlternativeFeerateSig(feerate, alternativeSig)
+                }
+                CommitSig(params.channelId, sig, listOf(), TlvStream(CommitSigTlv.AlternativeFeerateSigs(alternativeSigs)))
+            }
+            else -> CommitSig(params.channelId, sig, htlcSigs.toList())
+        }
```

Comment:
Nit: I would factor the declaration of `CommitSig()`:
```suggestion
        val tlvs = buildSet {
            if (spec.htlcs.isEmpty()) {
                val alternativeSigs = Commitments.alternativeFeerates.map { feerate ->
                    val alternativeSpec = spec.copy(feerate = feerate)
                    val (alternativeRemoteCommitTx, _) = Commitments.makeRemoteTxs(channelKeys, commitTxNumber = remoteCommit.index + 1, params.localParams, params.remoteParams, fundingTxIndex = fundingTxIndex, remoteFundingPubKey = remoteFundingPubkey, commitInput, remotePerCommitmentPoint = remoteNextPerCommitmentPoint, alternativeSpec)
                    val alternativeSig = Transactions.sign(alternativeRemoteCommitTx, channelKeys.fundingKey(fundingTxIndex))
                    CommitSigTlv.AlternativeFeerateSig(feerate, alternativeSig)
                }
                add(CommitSigTlv.AlternativeFeerateSigs(alternativeSigs))
            }
        }
        val commitSig = CommitSig(params.channelId, sig, htlcSigs.toList(), TlvStream(tlvs)) 
```

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/553/commits/abee6c6eb1fd120d49c653829380478973bf2ce7